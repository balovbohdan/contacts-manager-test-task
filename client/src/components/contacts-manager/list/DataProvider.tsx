import * as React from 'react';
import {isEmpty} from 'lodash';

type Props = {
    children;
    fetchContacts:()=>void;
};

export const DataProvider = ({children, fetchContacts}:Props) => {
    React.useEffect(() => fetchContacts(), []);

    return children;
};