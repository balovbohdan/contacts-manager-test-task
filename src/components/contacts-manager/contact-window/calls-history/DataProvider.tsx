import * as React from 'react';

import {FetchCallsHistory} from '@storehouse/actions/contacts/types';

type Props = {
    children;
    contactId:number;
    fetchCallsHistory:FetchCallsHistory;
};

export const DataProvider = (props:Props) => {
    const {contactId, children, fetchCallsHistory} = props;

    React.useEffect(() => fetchCallsHistory({ contactId }), []);

    return children;
};