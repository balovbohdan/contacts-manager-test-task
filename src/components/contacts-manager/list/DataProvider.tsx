import gql from 'graphql-tag';
import * as React from 'react';
import {useQuery} from '@apollo/react-hooks';

type Props = {
    children:React.ReactNode;
};

const QUERY = gql`
    query GetContacts {
        contacts {
            id
            name
            phone
        }
    }
`;

export const DataProvider = ({children}:Props) => {
    const {data, error, loading} = useQuery(QUERY);

    if (loading)
        return null;

    if (error)
        return <h1>GraphQL error happened...</h1>;

    console.log(data);

    return <>{children}</>;
};