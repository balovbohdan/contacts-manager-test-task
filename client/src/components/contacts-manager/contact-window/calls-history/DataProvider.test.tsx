import * as React from 'react';
import {mount, shallow} from 'enzyme';

import {DataProvider} from './DataProvider';

describe('CallsHistory data provider', () => {
    it('Should match snapshot', () => {
        const props = {
            contactId: 1,
            fetchCallsHistory: jest.fn()
        };

        const provider = shallow(<DataProvider {...props}>Foo</DataProvider>);

        expect(provider).toMatchSnapshot();
    });

    it('Should call data fetcher function', () => {
        const props = {
            contactId: 1,
            fetchCallsHistory: jest.fn()
        };

        mount(<DataProvider {...props}>Foo</DataProvider>);
        expect(props.fetchCallsHistory).toBeCalledTimes(1);
    });
});