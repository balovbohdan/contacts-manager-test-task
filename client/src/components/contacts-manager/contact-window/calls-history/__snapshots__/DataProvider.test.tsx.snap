// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallsHistory data provider Should match snapshot 1`] = `"Foo"`;
